lib/index.js

```diff
@@ -2,15 +2,85 @@
  * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
  */
 'use strict';
-// const {generateId} = require('bnid');
-// const {snakeCase} = require('snake-case');
-// const {
-//   InvalidRequest, AccessDenied
-// } = require('@interop/oauth2-errors');
 
-function handleTokenRequest({logger}) {
+const {
+  UnauthorizedClient, InvalidGrant, InvalidRequest, InvalidClient
+} = require('@interop/oauth2-errors');
+
+// Default Access token expiration in seconds (two weeks)
+const DEFAULT_MAX_AGE = 1209600;
+
+/**
+ * Constructs a middleware handler for OAuth 2.0 /token exchange requests.
+ *
+ * @param {object} options - Options.
+ * @param {string} options.baseUrl
+ * @param {object} [options.defaults={}]
+ * @param {object} logger
+ * @param {function} getClient - Callback to load client by id.
+ * @param {function} issue - Callback to issue the JWT access token for client.
+ *
+ * Usage:
+ * ```
+ * handleTokenExchange({
+ *   baseUrl: 'https://as.example.com',
+ *   defaults: {},
+ *   logger: console,
+ *   getClient: async ({clientId}) => {
+ *     // load client from the db, then:
+ *     return {client};
+ *   }),
+ *   issue: async ({client, scope, body, defaultMaxAge}) => {
+ *     // issue JWT access token here
+ *     return {accessToken, expiresIn};
+ *   })
+  },
+ * });
+ * ```
+ *
+ * @returns {Promise<function>} Resolves with exchange middleware.
+ */
+function handleTokenExchange({/*baseUrl,*/ defaults = {}, logger,
```

Comment:
Since this function doesn't actually handle a token exchange, we should name it something more like `createTokenExchangeHandler` or `tokenExchangeHandler` (if we are ok with a naming convention that drops the `create` for these middleware factories).

Reply:
+1, happy to rename to `tokenExchangeHandler`

Reply:
Renamed.